sycl/test-e2e/NonUniformGroups/tangle_group_algorithms.cpp

```diff
@@ -0,0 +1,137 @@
+// RUN: %clangxx -fsycl -fsycl-device-code-split=per_kernel -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+//
+// UNSUPPORTED: cpu || cuda || hip
+
+#include <sycl/sycl.hpp>
+#include <vector>
+namespace syclex = sycl::ext::oneapi::experimental;
+
+class TestKernel;
+
+constexpr uint32_t SGSize = 32;
+
+#define BRANCH_BODY()                                                          \
```

Comment:
Can it be a lambda instead? If not, we need a comment here. If yes, does it matter if we pass the `Tangle` by reference or create inside the body?

Reply:
I thought it wasn't possible (which is why I used a macro), but I clearly didn't try hard enough the first time :laughing:.

Fixed in 2045e6a.